### PR TITLE
mpd_source: clear current song info when refreshing

### PR DESCRIPTION
**Pull request by sim590**
Following the bug I mentioned [here](https://github.com/univrsal/tuna/issues/103#issuecomment-753540832), I did a fix.

It may not be really clean, I don't know if you meant to remove that line. @univrsal if you have any suggestion, please let me know.

### CODE REVIEW

**Comment by sim590** [comment]
Actually, after reading the code more in depth, I really think that you didn't mean to remove that line, but only wanted to remove the following in d2dfb6b69c211b61b4153c0e891d896fc52bcc38.

```cpp
#ifdef _WIN32
    song copy = m_current; /* Windows seems to randomly not return any info, so we keep a copy in case that happens */
#endif
```

It really seems like fields have to be cleared because we're doing appends on `QList`s.

File: src/query/mpd_source.cpp [hunk 1]
```diff
@@ -105,6 +105,7 @@ void mpd_source::refresh()
     struct mpd_connection* connection = connect();
     struct mpd_status* status = nullptr;
     struct mpd_song* mpd_song = nullptr;
+    m_current.clear();
```
Comment (univrsal) [inline src/query/mpd_source.cpp:108]: ``m_current.clear()`` should ideally be after ``begin_refresh()``, as ``begin_refresh()`` saves the current song for comparison.

**Comment by univrsal** [comment]
```C
#ifdef _WIN32
    song copy = m_current; /* Windows seems to randomly not return any info, so we keep a copy in case that happens */
#endif
```
Shouldn't be needed anymore as the connection isn't kept alive, but I don't have windows around to test it.

File: src/query/mpd_source.cpp (hunk 1 above)
Comment (sim590) [reply src/query/mpd_source.cpp]: Fixed.

**Comment by sim590** [comment]
> ```c
> #ifdef _WIN32
>     song copy = m_current; /* Windows seems to randomly not return any info, so we keep a copy in case that happens */
> #endif
> ```
> 
> Shouldn't be needed anymore as the connection isn't kept alive, but I don't have windows around to test it.

I don't have Windows either to test it... The safest would be to keep it, but if you're really confident that it shouldn't be needed, then I'm not opposing to letting it go :wink:.

**Pull request merged**